### ISSUE TITLE
Bug in function WiFiServer::write

### ISSUE THREAD

**Issue by MarkusLange** [arduino-libraries/WiFi101#160]
server.write function in WiFiServer.cpp does not work proper, function needs to change from

```
size_t WiFiServer::write(const uint8_t *buffer, size_t size)
{
	size_t n = 0;
	WiFiClient *client;

	for (int sock = 0; sock < TCP_SOCK_MAX; sock++) {
		client = WiFi._client[sock];
		if (client && client->_flag & SOCKET_BUFFER_FLAG_CONNECTED) {
			if (((client->_flag >> SOCKET_BUFFER_FLAG_PARENT_SOCKET_POS) & 0xff) == (uint8)_socket) {
				n += client->write(buffer, size);
			}
		}
	}
	return n;
}
```
to this

```
size_t WiFiServer::write(const uint8_t *buffer, size_t size)
{
	size_t n = 0;
	WiFiClient *client;

	for (int sock = 0; sock < TCP_SOCK_MAX; sock++) {
		client = WiFi._client[sock];
		if (client && client->_flag & SOCKET_BUFFER_FLAG_CONNECTED) {
			if (((client->_flag >> SOCKET_BUFFER_FLAG_PARENT_SOCKET_POS) & 0xff) == (uint8)(_socket + 1)) {
				n += client->write(buffer, size);
			}
		}
	}
	return n;
}
```
the socket needs to be added with 1 like its done in WiFiClient WiFiServer::available(uint8_t* status)

**Comment by sandeepmistry** [comment]
@MarkusLange thanks for brining this up, I've opened pull request https://github.com/arduino-libraries/WiFi101/pull/162 with the fix you suggested.

**Comment by sandeepmistry** [comment]
Closed via #162 which has been merged.
